operator/cmd/mesh/manifest-generate_test.go

```diff
@@ -160,6 +160,57 @@ func TestManifestGenerateComponentHubTag(t *testing.T) {
 	}
 }
 
+func TestManifestGenerateGateways(t *testing.T) {
+	testDataDir = filepath.Join(operatorRootDir, "cmd/mesh/testdata/manifest-generate")
+	g := NewGomegaWithT(t)
+	m, _, err := generateManifest("gateways", "", liveCharts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	objs := parseObjectSetFromManifest(t, m)
+	g.Expect(objs.kind(deploymentStr).size()).Should(Equal(3))
+	g.Expect(objs.kind(hpaStr).size()).Should(Equal(3))
+	g.Expect(objs.kind(pdbStr).size()).Should(Equal(3))
+	g.Expect(objs.kind(serviceStr).size()).Should(Equal(3))
+
+	// istio-ingressgateway and user-ingressgateway share these.
```

Comment:
Why is it 2 not 3?

Reply:
I would expect either 1 (they all share) or 3 (they all have their own)

Reply:
it's because there are two namespaces in the input spec. within each namespace, they are indeed shared. i've added some extra comment.